### ISSUE TITLE
Ignore words by regex

### ISSUE THREAD

**Issue by St-Ex-savadenn** [davidlday/vscode-languagetool-linter#150]
In the settings `Ignored Words Global` support both simple word and regular expression. If a word match either a word or a regex, it's ignored from the linter.

In the following exemple, the second entry would prevent several word to be detect as error : Github, Gitlab, Gitfoobarb, etc
![image](https://user-images.githubusercontent.com/65651015/85465929-a4f76a80-b5a9-11ea-836a-2b75c6f19a7a.png)

I'll try a PR with this idea if you want

_Originally posted by @St-Ex-savadenn in https://github.com/davidlday/vscode-languagetool-linter/issues/67#issuecomment-648439809_

**Comment by St-Ex-savadenn** [comment]
I just read [manager.ts](https://github.com/St-Ex-savadenn/vscode-languagetool-linter/blob/master/src/configuration/manager.ts) and it might be better to have a list of regular expression on its own. To prevent calling a regex match on a very long list of word.

**Comment by St-Ex-savadenn** [comment]
I was wondering if it's better to test all regex or to merge regex into one and test. 

## TL;DR;

Create the regex and cache it for best perf if we stay under 200 regex which is believable

## Tests

Here the results : 
1. For each word, tests all patterns
2. For each word, merge all patterns into one regex, test it 
3. Create regex, test it for each word
4. Using has (this is not a regex match, it's for compare perf)

| Words count | Patterns count | 1 | 2 | 3 |4|
| --------:|----:|---:|---:|---:|---:|
| 100 | 800 | 88 ms | 74 ms | 0 ms | 1 ms | 
| 100 | 900 | 84 ms | 89 ms | 1 ms | 0 ms | 
| 100 | 1000 | 87 ms | 102 ms | 1 ms | 0 ms | 
| 100 | 1100 | 108 ms | 102 ms | 1 ms | 0 ms | 
| 100 | 1200 | 115 ms | 132 ms | 0 ms | 1 ms | 
| 100 | 1300 | 140 ms | 159 ms | 112 ms | 0 ms | 
| 100 | 1400 | 122 ms | 132 ms | 0 ms | 0 ms | 
| 100 | 1500 | 111 ms | 432 ms | 74 ms | 0 ms | 
| 100 | 1600 | 136 ms | 476 ms | 76 ms | 0 ms | 
| 5000 | 10 | 15 ms | 18 ms | 6 ms | 0 ms | 
| 5000 | 20 | 30 ms | 30 ms | 15 ms | 0 ms | 
| 5000 | 50 | 85 ms | 87 ms | 43 ms | 0 ms | 
| 5000 | 100 | 162 ms | 183 ms | 89 ms | 0 ms | 
| 5000 | 200 | 415 ms | 410 ms | 202 ms | 0 ms | 
| 5000 | 500 | 1042 ms | 1369 ms | 850 ms | 0 ms | 
| 5000 | 1000 | 2200 ms | 3468 ms | 2422 ms | 0 ms | 

Le fichier de test 
```js
function rstring(length) {
  var result = "";
  var characters =
    "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";
  var charactersLength = characters.length;
  for (var i = 0; i < length; i++) {
    result += characters.charAt(Math.floor(Math.random() * charactersLength));
  }
  return result;
}

function testSuite(w, p) {
  let words = [];
  while (words.length < w) {
    words.push(rstring(Math.floor(Math.random() * 11) + 2));
  }

  let patterns = [];
  while (patterns.length < p) {
    patterns.push(
      rstring(Math.floor(Math.random() * 3) + 2) + ".*(" + rstring(1) + "{2})"
    );
  }

  function hasPatterns(word) {
    return patterns.includes(word);
  }

  function allPatterns(word) {
    for (let pattern of patterns) {
      if (new RegExp(pattern).test(word)) {
        return true;
      }
    }
    return false;
  }

  function mergePatterns(word) {
    return new RegExp("(" + patterns.join(")|(") + ")").test(word);
  }

  let regex = new RegExp("(" + patterns.join(")|(") + ")");
  function cacheRegex(word) {
    return regex.test(word);
  }

  function has(word) {
    return patterns.includes(word);
  }

  function test(callback) {
    let time = Date.now();
    for (const word of words) {
      callback(word);
    }
    return Date.now() - time;
  }

  console.log(
    "| " +
      words.length +
      " | " +
      patterns.length +
      " | " +
      test(allPatterns) +
      " ms | " +
      test(mergePatterns) +
      " ms | " +
      test(cacheRegex) +
      " ms | " +
      test(has) +
      " ms | "
  );
}

let nw = 100;
testSuite(nw, 800);
testSuite(nw, 900);
testSuite(nw, 1000);
testSuite(nw, 1100);
testSuite(nw, 1200);
testSuite(nw, 1300);
testSuite(nw, 1400);
testSuite(nw, 1500);
testSuite(nw, 1600);

nw = 5000;
testSuite(nw, 10);
testSuite(nw, 20);
testSuite(nw, 50);
testSuite(nw, 100);
testSuite(nw, 200);
testSuite(nw, 500);
testSuite(nw, 1000);
```
